src/field.rs

```diff
@@ -190,6 +196,55 @@ pub trait FieldElement:
         }
         Ok(vec)
     }
+
+    /// Encode `input` as `bits`-bit vector of elements of `Self` if it's small enough
+    /// to be represented with that many bits.
+    ///
+    /// # Arguments
+    ///
+    /// * `input` - The field element to encode
+    /// * `bits` - The number of bits to use for the encoding
+    /// * `max_input` - The maximal number representable with `bits` bits. Equal to `bits^2-1`
+    fn encode_into_bitvector_representation(
+        input: &Self::Integer,
+        bits: usize,
+        max_input: Self::Integer,
+    ) -> Result<Vec<Self>, FieldError> {
+        if *input > max_input {
+            return Err(FieldError::InputSizeMismatch);
+        }
+
+        let one = Self::Integer::from(Self::one());
+        let mut encoded = Vec::with_capacity(bits);
+        for l in 0..bits {
+            let l = Self::Integer::try_from(l).map_err(|_| {FieldError::IntegerTryFrom})?;
+            let w = Self::from((*input >> l) & one);
+            encoded.push(w);
+        }
+        Ok(encoded)
+    }
+
+    /// Decode the bitvector-represented value `input` into a simple representation as a single field element.
+    /// This function panics if `2 ^ input.len()` does not fit into the field `Self`.
+    fn decode_from_bitvector_representation(input: &[Self]) -> Self {
+        let mut decoded = Self::zero();
+        for (l, bit) in input.iter().enumerate() {
+            let w = Self::from(Self::Integer::try_from(1 << l).unwrap());
+            decoded += w * *bit;
+        }
+        decoded
+    }
+
+    /// Interpret `i` as [`Self::Integer`] if it's representable in that type and smaller than the field modulus.
+    fn valid_integer_try_from<N>(i: N) -> Result<Self::Integer, FieldError> where Self::Integer: TryFrom<N>{
```

Comment:
Consider removing the `valid_` prefix here. It' doesn't make sense in this context.

Reply:
The idea was that this is not simply a value of type `Field::Integer` but one that is smaller than the modulus, thus a "representable integer" or "valid integer". But we can remove that prefix.

Reply:
Ack, you can leave as is.